301-2fe-lb80-rdp-1be-nsg-rdp/azuredeploy.json

```diff
@@ -0,0 +1,389 @@
+﻿{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "publicDnsName": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique public DNS prefix for the deployment. The fqdn will look something like '<dnsname>.westus.cloudapp.azure.com'. Up to 62 chars, digits or dashes, lowercase, should start with a letter: must conform to '^[a-z][a-z0-9-]{1,61}[a-z0-9]$'."
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the administrator of the new VM. Exclusion list: 'admin','administrator'"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The password for the administrator account of the new VM"
+      }
+    }
+  },
+  "variables": {
+    "vnetAddressRange": "10.0.0.0/16",
+    "subnetAddressRange": "10.0.0.0/24",
+    "subnetName": "Subnet",
+    "numberOfInstances": 2,
+    "availabilitySetName": "myavlset",
+    "vmName" : "vm",
+    "nicsql": "[concat(variables('vmName'),'sql')]",
+    "newStorageAccountName": "[concat(parameters('publicDnsName'),'st')]",
```

Comment:
this will fail in case someone enters a capital letter for dns names. best to get this as input or use the new uniqueString() function in ARM https://azure.microsoft.com/en-us/documentation/articles/resource-group-template-functions/#uniquestring